rasa/core/tracker_store.py

```diff
@@ -349,25 +349,31 @@ def serialise_tracker(self, tracker: "DialogueStateTracker") -> Dict:
                 "session_date": int(datetime.now(tz=timezone.utc).timestamp()),
             }
         )
+        # DynamoDB cannot store `float`s, so we'll convert them to `Decimal`s
         return utils.replace_floats_with_decimals(d)
 
     def retrieve(self, sender_id: Text) -> Optional[DialogueStateTracker]:
         """Create a tracker from all previously stored events."""
-
-        # Retrieve dialogues for a sender_id in reverse chronological order based on
+        # Retrieve dialogues for a sender_id in reverse-chronological order based on
         # the session_date sort key
         dialogues = self.db.query(
             KeyConditionExpression=Key("sender_id").eq(sender_id),
             Limit=1,
             ScanIndexForward=False,
         )["Items"]
-        if dialogues:
-            return DialogueStateTracker.from_dict(
-                sender_id, dialogues[0].get("events"), self.domain.slots
-            )
-        else:
+
+        if not dialogues:
             return None
 
+        events = dialogues[0].get("events", [])
+
+        # `float`s are stored as `Decimal` objects - we need to convert them back
+        events_with_floats = utils.replace_decimals_with_floats(events)
```

Comment:
Minor: Should we import `utils` as `core_utils` to clarify where those methods come from?

Reply:
👍